pkg/ccl/crosscluster/producer/replication_manager.go

```diff
@@ -207,7 +214,14 @@ func (r *replicationStreamManagerImpl) PlanLogicalReplication(
 			return nil, err
 		}
 	}
-	spec, err := buildReplicationStreamSpec(ctx, r.evalCtx, tenID, false, spans, useStreaksInLDR.Get(&r.evalCtx.Settings.SV))
+
+	// Pass targetSpans = 1 to prevent re-partitioning spans.
+	targetSpans := 1
```

Comment:
nit: call this variable `targetPartitionCount`? The current varname is quite confusing as one var in the `buildReplicationStreamSpec`signature is `targetSpans` and this var is passed to `targetPartitionCount`, not `targetSpans`

Reply:
made it a setting actually